pegasus/sites.v3/code.org/public/edit-csp-table/records.haml

```diff
@@ -93,9 +113,9 @@ content-type: text/ng-template
           {{ record.id }}
       %td{"ng-repeat" => "column in columns"}
         %span{"ng-hide" => "record.editing"}
-          {{ record[column] }}
+          {{ record[column.id] }}
         %span{"ng-show" => "record.editing"}
-          %input{"ng-model" => "record[column]", type: "text"}/
+          %input{"ng-model" => "record[column.name]", type: "text"}/
```

Comment:
Good bug! Just pushed the fix.

On Fri, Sep 11, 2015 at 2:53 PM, davidsbailey <notifications@github.com>
wrote:

> In pegasus/sites.v3/code.org/public/edit-csp-table/records.haml
> <https://github.com/code-dot-org/code-dot-org/pull/3945#discussion_r39321953>
> :
>
> >          %span{"ng-show" => "record.editing"}
> > -          %input{"ng-model" => "record[column]", type: "text"}/
> > +          %input{"ng-model" => "record[column.name]", type: "text"}/
>
> I did notice that if I have a row open for edit, and I have a column name
> open for edit, and I start to change the column name, that the value in the
> intersection of that row/column disappears:
>
> [image: screen shot 2015-09-11 at 2 48 16 pm]
> <https://cloud.githubusercontent.com/assets/8001765/9827169/8a8b8bb2-5894-11e5-862b-d3d1ff9cf79f.png>
>
> Saving the column name to something new, or leaving the column name open
> for edit but returning it to its original text, each make the blank cell
> reappear. changing column.name to column.id here seems like it should fix
> it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/3945/files#r39321953>.
>



-- 
Trevor
trevor@code.org